HtmlMesh/src/html-mesh-renderer.ts

```diff
@@ -214,18 +216,21 @@ export class HtmlMeshRenderer {
             );
         }
 
+        const { width, height } = scene.getEngine().getRenderingCanvasClientRect() ?? _defaultRenderingSize;
```

Comment:
Getting the rect can be an expensive operation.  I created functions to get a memoized client rect in `util.js`, `getCanvasRectAsync` and `getCanvasRectOrNull`.  The former will retry if the canvas rect is not yet defined whereas the latter will just return null.  Could you please update your PR to use one of these functions?